[13.0][MIG] account_tax_balance

astirpe opened the pull request:
Depends on:

- [x] https://github.com/OCA/server-ux/pull/95

Includes:

- [ ] https://github.com/OCA/account-financial-reporting/pull/613

eLBati committed:
[ADD] account_tax_balance: Compute tax balances based on date range

gfcapalbo committed:
tests

eLBati committed:
[FIX+IMP] account_tax_balance:

* Tests
* PEP8
* Use invoice._convert_to_write(invoice._cache). This way, the onchange will be inheritable and will add here also the added values
* better get_context_values
* unify method compute_balance
* open move lines linked to balance

antespi committed:
Consider normal and refund operations separately

Allow to explore all move lines

Spanish translation

Bugfixes

Show negative lines too

Show move type in account.move views

Tests include new fields

sbidoul committed:
[MIG] account_tax_balance 10.0

eLBati committed:
[FIX+IMP] account_tax_balance:

* Test
* README
* add extra parent menu. If not, with the web_responsive module is mixed
  between other menus

oca-transbot committed:
OCA Transbot updated translations from Transifex

StefanRijnhart committed:
[FIX] Include tax amounts from all move types

sbidoul committed:
[IMP] account_tax_balance: performance

Let the database do some computations (sum) and do not
put large lists of ids in action domain.

sbidoul committed:
[IMP] account_tax_balance: faster search of taxes with moves

sbidoul committed:
account_tax_balance: improve test coverage

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
add condition tax_exigible in domain variable

sbidoul committed:
[IMP] account_tax_balance: add index

This indexes improve the display time of
account_tax_balance by a factor of 10 on a database
with 1M move lines.

pedrobaeza committed:
[MIG] account_tax_balance: Migration to 11.0

JordiBForgeFlow committed:
dynamize general ledger

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[IMP] account_tax_balance: Change tests to post instal and protect again freezes

oca-transbot committed:
OCA Transbot updated translations from Transifex

HviorForgeFlow committed:
utf-8 remove from .py files

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-travis committed:
[UPD] Update account_tax_balance.pot

dcorio committed:
Translated using Weblate (Italian)

Currently translated at 9.8% (5 of 51 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_tax_balance
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_tax_balance/it/

jimmybanegas committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (51 of 51 strings)

Translation: account-financial-reporting-11.0/account-financial-reporting-11.0-account_tax_balance
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-11-0/account-financial-reporting-11-0-account_tax_balance/es/

astirpe committed:
[12.0][MIG] account_tax_balance

OCA-git-bot committed:
[UPD] README.rst

astirpe committed:
[FIX] at_install/post_install test decorators

JordyBlankestijn committed:
Translated using Weblate (Dutch)

Currently translated at 47.1% (24 of 51 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-account_tax_balance
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-account_tax_balance/nl_NL/

OCA-git-bot committed:
[UPD] README.rst

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 80.4% (41 of 51 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-account_tax_balance
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-account_tax_balance/it/

oca-travis committed:
[UPD] Update account_tax_balance.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 84.8% (39 of 46 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-account_tax_balance
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-account_tax_balance/it/

mvrodriguez committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (46 of 46 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-account_tax_balance
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-account_tax_balance/es/

BECJIE32 committed:
Translated using Weblate (French)

Currently translated at 95.7% (44 of 46 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-account_tax_balance
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-account_tax_balance/fr/

pedrobaeza committed:
[IMP] account_tax_balance: Ensure test correctness

On certain environments, expense account might not exist, so we create it directly
for avoiding problems.

pedrobaeza committed:
[IMP] account_tax_balance: Ensure test correctness (2)

Tests failed if you don't have an account called 'Tax Paid' in your test DB, which
can happen when you are integrating tests in other environments, for example with
other starting language.

bealdav committed:
[IMP] tax_balance: breadcrumb name (#548)

oca-travis committed:
[UPD] Update account_tax_balance.pot

Commit:
Translated using Weblate (Catalan)

Currently translated at 67.4% (31 of 46 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-account_tax_balance
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-account_tax_balance/ca/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-account_tax_balance
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-account_tax_balance/

josepmy committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (47 of 47 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-account_tax_balance
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-account_tax_balance/es/

OCA-git-bot committed:
[UPD] README.rst

astirpe committed:
[IMP] Ease extension if multicompany fiscal unit

astirpe committed:
[13.0][MIG] account_tax_balance

astirpe committed:
Convert onchange to computed

astirpe committed:
Code review

feketemihai commented:
@astirpe There is a good report over tax directly on v13 in Odoo base that could be configured more based on tags, countries..https://github.com/odoo/odoo/blob/master/addons/account/models/account.py#L62
Can we just adapt it?

astirpe commented:
@feketemihai how do you propose to "adapt it"?
This module never used tags in previous versions.

feketemihai commented:
@astirpe I was thinking if you can use the report to get the computed fields

astirpe commented:
IMO this module is meant to do something different than what `account.tax.report.line` does.

pedrobaeza commented:
Yes, I think both are needed.

astirpe committed:
Add oca_dependencies.txt

astirpe committed:
Black, isort, etc...

StefanRijnhart committed:
[ADD] Prefill move_type

Precreate move_type and fill with appropriate values to prevent
a MemoryError when the ORM attempts to call its compute method on a large
amount of preexisting moves.

MiquelRForgeFlow commented:
Is it still wip?

astirpe commented:
This is ready for being reviewed IMO

pedrobaeza committed:
[RFR] Apply a mapping instead of separate queries

MiquelRForgeFlow commented on account_tax_balance/tests/test_account_tax_balance.py:
```diff
@@ -183,13 +184,14 @@ def test_tax_balance(self):
                 'credit': 10,
                 'name': 'Bank Fees',
-                'tax_line_id': tax.id,
+                'tax_repartition_line_id': tax_repartition_line.id,
```
I don't know why you change this line. I think it should be as before. 🤔

astirpe replied:
Because [ `tax_line_id` is a compute field](https://github.com/odoo/odoo/blob/13.0/addons/account/models/account_move.py#L2454-L2455) and doesn't have any method for the inverse.

MiquelRForgeFlow replied:
:+1:

MiquelRForgeFlow reviewed (approved):
code review LGTM

rafaelbn reviewed (approved):
Ready to merge by a PSC :+1:

pedrobaeza commented:
@astirpe can you please cherry-pick #647 for performance reasons on install?

pedrobaeza reviewed (changes requested):
I have put comments in #613, and for v13, Odoo has changed the way multi-companies work (now you have `self.env.company` and `self.env.companies`), and you should change according this the queries you were modifying in that PR.

astirpe commented:
@pedrobaeza the list of companies ids is passed by context from any localization (or custom) module extending this one. The list of companies depends on how the localization module is supposed to work.

See for example https://github.com/OCA/l10n-netherlands/issues/237 and its proposed solution https://github.com/OCA/l10n-netherlands/pull/238/files.
As you can notice, the list of companies to be reported is calculated and can be different from the classic parent-child. In our module it also depends on the country and the currency.

I would keep the code as it is.
If you don't like this way, I could remove this change from this PR and move it to a separate PR. At least we don't block the porting.

pedrobaeza commented:
/ocabot merge
